utilities/ideascale-importer/ideascale_importer/snapshot_importer/importer.py

```diff
@@ -103,7 +105,7 @@ async def _fetch_parameters(self):
             )
 
             row = await conn.fetchrow(
-                "SELECT slot_no FROM block WHERE time <= $1 ORDER BY time DESC LIMIT 1",
+                "SELECT time, slot_no FROM block WHERE time <= $1 ORDER BY time DESC LIMIT 1",
```

Comment:
On reflection, for the latest slot_no and time, I think the query should just be:
```sql
SELECT time, slot_no FROM block SORT BY slot_no DESC LIMIT 1;
```
Which will give you the latest slot number and time in dbsync.

This query is still valid for getting the slot number the snapshot needs to be run as at.

So you need two slot numbers:
1. The "as at" slot number which is when we align the registrations and voting power.
2. The "current" slot number which is the latest time and slot number recorded by dbsync.

Final is only related to the second.

Also, before running the snapshot, we should check IF the latest snapshot for the event in the DB is already marked final and not re-run it if it is.